sourceCpp compile error stderr cannot be caught/suppressed

When writing a `try()` function to check for compile errors from `sourceCpp` the compilation error is always printed to the console, regardless of whichever combinations of `try()`, `suppressMessages()`, `suppressWarnings()` or `sink` are used.

Reprex:

``` r
dcode <- "//[[Rcpp::export]]\n int fun(int x) { const int y; return x; }"
status <- suppressWarnings(suppressMessages(
  output <- capture.output(
    try(Rcpp::sourceCpp(code = dcode, verbose=FALSE, echo=FALSE),
                 silent = TRUE))
))
```

<sup>Created on 2023-03-24 with [reprex v2.0.2](https://reprex.tidyverse.org)</sup>

<summary>
Standard output and standard error
</summary>

``` sh
fileeed07e197428.cpp:2:29: error: default initialization of an object of const type 'const int'
 int fun(int x) { const int y; return x; }
                            ^
                              = 0
1 error generated.
make: *** [fileeed07e197428.o] Error 1
```

--- a/ChangeLog
+++ b/ChangeLog
@@ -1,3 +1,7 @@
+2023-03-25  Iñaki Ucar  <iucar@fedoraproject.org>
+
+	* R/Attributes.R: Switch to system2 to be able to capture stderr on error
+
 2023-03-24  Dirk Eddelbuettel  <edd@debian.org>
 
 	* docker/ci-dev/Dockerfile: During Debian freeze experimental repo is

--- a/R/Attributes.R
+++ b/R/Attributes.R
@@ -137,7 +137,7 @@ sourceCpp <- function(file = "",
 
         # prepare the command (output if we are in showOutput mode)
         args <- c(
-            r, "CMD", "SHLIB",
+            "CMD", "SHLIB",
             if (windowsDebugDLL) "-d",
             if (rebuild) "--preclean",
             if (dryRun) "--dry-run",
@@ -147,13 +147,13 @@ sourceCpp <- function(file = "",
             shQuote(src)
         )
 
-        cmd <- paste(args, collapse = " ")
         if (showOutput)
-            cat(cmd, "\n")										# #nocov
+            cat(paste(c(r, args), collapse = " "), "\n")		# #nocov
 
         # execute the build -- suppressWarnings b/c when showOutput = FALSE
         # we are going to explicitly check for an error and print the output
-        result <- suppressWarnings(system(cmd, intern = !showOutput))
+        so <- if (showOutput) "" else TRUE
+        result <- suppressWarnings(system2(r, args, stdout = so, stderr = so))
 
         # check build results
         if(!showOutput) {

